src/main/java/com/mraof/minestuck/item/weapon/CustomBow.java

```diff
@@ -0,0 +1,133 @@
+package com.mraof.minestuck.item.weapon;
+
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.enchantment.Enchantments;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.AbstractArrowEntity;
+import net.minecraft.item.*;
+import net.minecraft.stats.Stats;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.world.World;
+
+public class CustomBow extends BowItem
+{
+	private final Type type;
+	
+	public enum Type
+	{
+		FLAME, LONG, NO_GRAVITY, QUICK, RANDOM_SECOND
+	}
+	
+	public CustomBow(Item.Properties builder, Type type)
+	{
+		super(builder);
+		this.type = type;
+		this.addPropertyOverride(new ResourceLocation("pull"), (p_210310_0_, p_210310_1_, p_210310_2_) -> {
+			if(p_210310_2_ == null)
+			{
+				return 0.0F;
+			} else
+			{
+				return !(p_210310_2_.getActiveItemStack().getItem() instanceof CustomBow) ? 0.0F : (float) (p_210310_0_.getUseDuration() - p_210310_2_.getItemInUseCount()) / 20.0F;
+			}
+		});
+		this.addPropertyOverride(new ResourceLocation("pulling"), (p_210309_0_, p_210309_1_, p_210309_2_) -> {
+			return p_210309_2_ != null && p_210309_2_.isHandActive() && p_210309_2_.getActiveItemStack() == p_210309_0_ ? 1.0F : 0.0F;
+		});
+	}
+	
+	@Override
+	public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity entityLiving, int timeLeft)
+	{
+		if(entityLiving instanceof PlayerEntity)
+		{
+			PlayerEntity playerentity = (PlayerEntity) entityLiving;
+			boolean canShootFlag = playerentity.abilities.isCreativeMode || EnchantmentHelper.getEnchantmentLevel(Enchantments.INFINITY, stack) > 0;
+			ItemStack itemstack = playerentity.findAmmo(stack);
+			
+			int i = this.getUseDuration(stack) - timeLeft;
+			i = net.minecraftforge.event.ForgeEventFactory.onArrowLoose(stack, worldIn, playerentity, i, !itemstack.isEmpty() || canShootFlag);
+			if(i < 0) return;
```

Comment:
Because it is the default code, its readability needs improving, will change the name of these undescriptive variables